### PR TITLE
Better type names in metadata

### PR DESCRIPTION
**Pull request by triggerNZ**

### CODE REVIEW

File: core/src/main/scala/commbank/coppersmith/Feature.scala [hunk 1]
```diff
@@ -143,7 +143,21 @@ object Feature {
     }
     object TypeInfo {
       def apply[T : TypeTag]: TypeInfo = TypeInfo(implicitly[TypeTag[T]].tpe)
-      def apply(t: ScalaType): TypeInfo = TypeInfo(t.typeSymbol.fullName, t.typeArgs.map(TypeInfo(_)))
+      def apply(t: ScalaType): TypeInfo = {
```
Comment (dkristian) [inline core/src/main/scala/commbank/coppersmith/Feature.scala]: I would prefer this logic to be externalised if possible, eg, moved to `MetadataOutput`.

There's also a bit of repetition in the logic. How about:
```scala
name.replaceAll("^java.lang.", "")
  .replaceAll("^scala.collection.immutable.", "")
  .replaceAll("^scala.", "")
```

**Comment by dkristian** [comment]
Missing version bump.

File: core/src/main/scala/commbank/coppersmith/Feature.scala (hunk 1 above)
Comment (triggerNZ) [inline core/src/main/scala/commbank/coppersmith/Feature.scala]: Okay, good point. I will move to Output. I avoided `replaceAll` as a performance habit I guess, regex compilation is expensive. When generating the small number of features that our jobs tend to have, I think it doesn't matter

Comment (dkristian) [inline core/src/main/scala/commbank/coppersmith/Feature.scala]: I had wondered if you were consciously steering clear of regexs. Normally I would too, but this seems relatively simple (and yes, I realise that the `.` chars above are interpreted as *any* character, but figure it's pretty safe under these circumstances).

Another option would be
```scala
val prefixesToDrop = List("java.lang.", "scala.collection.immutable.", "scala.")
prefixesToDrop.foldLeft(name)((n, prefix) =>
  if (name.startsWith(prefix)) n.substring(prefix.length) else n
)
```
but I'm aware I might be heading into golfing/bike-shedding territory. My main objection with the original code is the magic numbers.

**Comment by triggerNZ** [comment]
Better addressed with the published JSON PR coming up

**Pull request closed without merging**